kubernetes/client.go

```diff
@@ -68,26 +68,32 @@ func (c *Client) Write(p []byte) (int, error) {
 var ErrInterrupt = errors.New("interrupt signal received")
 
 func (c *Client) Await(ctx context.Context, desiredState RunState) error {
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	first := make(chan struct{}, 1)
+	first <- struct{}{}
 	for {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
-		default:
-			var current RunState
-			if err := c.client.Call("Runner.Status", c.ID, &current); err != nil {
-				return err
-			}
-			if current == desiredState {
-				return nil
-			}
-			if current == RunStateInterrupt {
-				return ErrInterrupt
-			}
-			select {
-			case <-ctx.Done():
-				return ctx.Err()
-			case <-time.After(time.Second):
-			}
+
+		case <-first:
```

Comment:
whats with the first thing?

Reply:
`time.Ticker` doesn't tick until after the duration, but we want to call the RPC immediately. `first` provides a path through the select that isn't blocked on the first iteration, but is blocked forever afterwards.

Reply:
Nifty, trick, so it is like "once" or "immediate", I get flash back to `do while` 😂

Reply:
@DrJosh9000 Maybe add a comment just to make this obvious?

Reply:
Done.